packages/core-data/src/selectors.ts

```diff
@@ -365,7 +365,7 @@ export const getEntityRecord = createSelector(
 		}
 		const context = query?.context ?? 'default';
 
-		if ( query === undefined ) {
+		if ( ! query || ! query._fields ) {
```

Comment:
Does this now make the `&& query._fields` on the condition below redundant? 

https://github.com/WordPress/gutenberg/blob/0e2eb9fd046f516f978f7cae36889ddd94838982/packages/core-data/src/selectors.ts#L378

Reply:
You're right. I don't have a strong opinion here, but it seems like a good check to keep in place "just in case" 😄

P.S. Happy to remove it if it's a blocker.